crypto/asn1/ameth_lib.c

```diff
@@ -124,24 +125,26 @@ const EVP_PKEY_ASN1_METHOD *EVP_PKEY_asn1_find_str(ENGINE **pe,
 #endif
         *pe = NULL;
     }
-    for (i = 0; i < EVP_PKEY_asn1_get_count(); i++) {
+    for (i = EVP_PKEY_asn1_get_count(); i-- > 0; ) {
```

Comment:
yay!  side-note I prefer ```--i >= 0``` YMMV.

Reply:
I don't, obviously ;-)

Reply:
For predefined methods that is probably fine,
but that the app_methods are sorted, what is the sort order?

Reply:
The sort order is by NID (see the sorting being done in `EVP_PKEY_asn1_add0`)

Reply:
I am not sure what will happen if a NID is replaced.
Is the sort function stable? will the new one be at higher index?

Reply:
It really isn't in the interest of the application to add several methods with the same base NID.  So hmm, perhaps we should make sure that it's not added if there's already with same base NID in `app_methods`...

Reply:
yes, that would be better.

Reply:
Okie, doing so.